src/matrix_multiply_add.jl

```diff
@@ -0,0 +1,263 @@
+
+# 5-argument matrix multiplication
+@inline LinearAlgebra.mul!(dest::StaticVecOrMat, A::StaticMatrix, B::StaticVecOrMat, α::Real, β::Real) =
+    _mul!(Size(dest), dest, Size(A), Size(B), A, B, α, β)
+@inline mul!(dest::StaticVecOrMat, A::StaticVector, B::Transpose{<:Any, <:StaticVector}, α::Real, β::Real) =
+    _mul!(Size(dest), dest, Size(A), Size(B), A, B, α, β)
+@inline mul!(dest::StaticVecOrMat, A::StaticVector, B::Adjoint{<:Any, <:StaticVector}, α::Real, β::Real) =
+    _mul!(Size(dest), dest, Size(A), Size(B), A, B, α, β)
+
+@inline mul!(dest::StaticVector, A::Transpose{<:Any, <:StaticMatrix}, B::StaticVector, α::Real, β::Real) =
+    _tmul!(Size(dest), dest, Size(A.parent), Size(B), A.parent, B, α, β)
+@inline mul!(dest::StaticMatrix, A::Transpose{<:Any, <:StaticMatrix}, B::StaticMatrix, α::Real, β::Real) =
+    _mul!(Size(dest), dest, Size(A), Size(B), A, B, α, β)
```

Comment:
Is it possible to arrange this dispatch a bit more cleanly rather than having quite so much code duplication here?

For example, you can define a type alias similar to `StaticMatrixLike` which captures both `StaticMatrix`, `Transpose` (and possibly `Adjoint`).

Then a single overload of `mul!` may be sufficient.